src/USER-MANIFOLD/manifold_gaussian_bump.cpp

```diff
@@ -1,138 +1,374 @@
 #include "manifold_gaussian_bump.h"
 
+#include "comm.h"
+#include "error.h"
+
 using namespace LAMMPS_NS;
 using namespace user_manifold;
 
+// This manifold comes with some baggage;
+// it needs a taper function that is sufficiently smooth
+// so we use a cubic Hermite interpolation and then for
+// efficiency we construct a lookup table for that....
+
+class cubic_hermite
+{
+public:
+  // Represent x-polynomial as  a * t^3 + b * t^2 + c*t + d.
+  double a, b, c, d;
+  // And y-polynomial as s * t^3 + u * t^2 + v * t + w.
+  double s, u, v, w;
+
+  double x0, x1, y0, y1, yp0, yp1;
+
+  LAMMPS_NS::Error *err;
+
+  cubic_hermite( double x0, double x1, double y0, double y1,
+                 double yp0, double yp1, LAMMPS_NS::Error *err ) :
+    x0(x0), x1(x1), y0(y0), y1(y1), yp0(yp0), yp1(yp1),
+    a(  2*x0 + 2 - 2*x1 ),
+    b( -3*x0 - 3 + 3*x1 ),
+    c( 1.0 ),
+    d( x0 ),
+    s(  2*y0 - 2*y1 +   yp0 + yp1 ),
+    u( -3*y0 + 3*y1 - 2*yp0 - yp1  ),
+    v(  yp0  ),
+    w(  y0 ),
+    err(err)
+  {
+    test();
+  }
+
+
+  void test()
+  {
+    if( fabs( x(0) - x0 ) > 1e-8 ) err->one(FLERR, "x0 wrong");
+    if( fabs( x(1) - x1 ) > 1e-8 ) err->one(FLERR, "x1 wrong");
+    if( fabs( y(0) - y0 ) > 1e-8 ) err->one(FLERR, "y0 wrong");
+    if( fabs( y(1) - y1 ) > 1e-8 ) err->one(FLERR, "y1 wrong");
+  }
+
+  double get_t_from_x( double xx ) const
+  {
+    if( xx < x0 || xx > x1 ){
+      char msg[2048];
+      sprintf(msg, "x ( %g ) out of bounds [%g, %g]", xx, x0, x1 );
+      err->one(FLERR, msg);
+    }
+
+    // Newton iterate to get right t.
+    double t  = xx - x0;
+    double dx = x1 - x0;
+    // Reasonable initial guess I hope:
+    t /= dx;
+
+    double tol = 1e-8;
+    int maxit = 500;
+    double ff  = x(t) - xx;
+    double ffp = xp(t);
+    // double l   = 1.0 / ( 1 + res*res );
+    for( int i = 0; i < maxit; ++i ){
+      t -= ff / ffp;
+      ff  = x(t) - xx;
+      ffp = xp(t);
+      double res = ff;
+      if( std::fabs( res ) < tol ){
```

Comment:
This prevents compilation on some systems. `fabs()` is not in `std::` if imported from C-style headers. Just use `fabs()` directly to stay compatible.